### PR TITLE
Add quic_bug_tracker_impl.h to QUICHE platform implementation.

### PR DESCRIPTION
**Pull request by wu-bin**
Description:

Add quic_bug_tracker_impl.h to QUICHE platform implementation.

(Also removed quiche/quic/platform/api/quic_interval.h from bazel/external/quiche.BUILD since it has been de-platformized)

Risk Level: none; code not used yet.
Testing:

bazel test test/extensions/quic_listeners/quiche/platform:quic_platform_test --test_output=all --define quiche=enabled --experimental_remap_main_repo

Docs Changes: none
Release Notes: none
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc [hunk 1]
```diff
@@ -45,6 +46,14 @@ TEST(QuicPlatformTest, QuicArraysize) {
 
 enum class TestEnum { ZERO = 0, ONE, TWO, COUNT };
 
+TEST(QuicPlatformTest, QuicBugTracker) {
+  // Just make sure they compile.
+  QUIC_BUG << "Here is a bug";
```
Comment (danzh2010) [inline test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: For my own knowledge, why does QUIC_BUG not abort here?

Comment (wu-bin) [reply test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: Nothing surprising here, it does abort in debug mode. My intention is just to check these statements can compile, please take a look at the latest commit.

Sorry for the stupidness. I must have context switched between writing this test and creating the PR.

File: test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc [hunk 2]
```diff
@@ -45,6 +46,17 @@ TEST(QuicPlatformTest, QuicArraysize) {
 
 enum class TestEnum { ZERO = 0, ONE, TWO, COUNT };
 
+TEST(QuicPlatformTest, QuicBugTracker) {
+  if (false) {
```
Comment (danzh2010) [inline test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: Instead of doing this, can you give a try to use `'EXPECT_DEATH'

Comment (wu-bin) [reply test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: Done. PTAL.

File: test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc [hunk 3]
```diff
@@ -45,6 +46,13 @@ TEST(QuicPlatformTest, QuicArraysize) {
 
 enum class TestEnum { ZERO = 0, ONE, TWO, COUNT };
 
+TEST(QuicPlatformTest, QuicBugTracker) {
+  EXPECT_DEBUG_DEATH(QUIC_BUG << "Here is a bug,", "");
+  EXPECT_DEBUG_DEATH(QUIC_BUG_IF(true) << "There is a bug,", "");
```
Comment (danzh2010) [inline test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: Nice find!

**Comment by wu-bin** [comment]
Ping. @alyssawilk @mattklein123

File: test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc [hunk 4]
```diff
@@ -48,6 +49,13 @@ TEST(QuicPlatformTest, QuicArraysize) {
...
+  QUIC_PEER_BUG << "Everywhere's a bug,";
```
Comment (alyssawilk) [inline test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: expect log contains macro checks?

Also checks for if(false) not happening would be good.

Comment (wu-bin) [reply test/extensions/quic_listeners/quiche/platform/quic_platform_test.cc]: Done. PTAL.

**Pull request merged**